Apollo/Components/FadeThumb.cs

```diff
@@ -47,44 +67,87 @@ public class FadeThumb: UserControl {
 
         bool dragged = false;
 
-        void DragStarted(object sender, VectorEventArgs e) {
+        void DragStarted(object sender, VectorEventArgs e)
+        {
             ((Window)this.GetVisualRoot()).Focus();
             dragged = false;
         }
 
-        void DragCompleted(object sender, VectorEventArgs e) {
+        void DragCompleted(object sender, VectorEventArgs e)
+        {
             if (!dragged) Focused?.Invoke(this);
             else if (change != 0) Moved?.Invoke(this, 0, change);
         }
 
-        void MouseDown(object sender, PointerPressedEventArgs e) {
+        void MouseDown(object sender, PointerPressedEventArgs e)
+        {
             PointerUpdateKind MouseButton = e.GetCurrentPoint(this).Properties.PointerUpdateKind;
 
             if (MouseButton != PointerUpdateKind.LeftButtonPressed) e.Handled = true;
-            
+
             ((Window)this.GetVisualRoot()).Focus();
         }
 
-        void MouseUp(object sender, PointerReleasedEventArgs e) {
+        void MouseUp(object sender, PointerReleasedEventArgs e)
+        {
             PointerUpdateKind MouseButton = e.GetCurrentPoint(this).Properties.PointerUpdateKind;
 
-            if (MouseButton == PointerUpdateKind.RightButtonReleased) {
-                Deleted?.Invoke(this);
+            if (MouseButton == PointerUpdateKind.RightButtonReleased)
+            {
+                MenuItem selectedItem = null;
+                foreach (Control item in ThumbContextMenu.Items)
+                {
+                    if (item.GetType() != typeof(MenuItem)) break;
+                    MenuItem menuItem = (MenuItem)item;
+                    menuItem.Icon = "";
+                    if (menuItem.Header.ToString() == FadeType.ToString())
+                    {
+                        selectedItem = menuItem;
```

Comment:
can't you just set the selectedicon here and break? since the menuitems are all the same, can you convert the enum to an int and just set the icon for the desired menuitem in one line and skip this whole loop?

Reply:
Since ThumbContextMenu.Items is an IEnumerable, I have to loop through all the items, set their icon to null, and then for the item that is selected set the icon to the indicator. I would have liked to just set the icon to the item of the required index, but I'm not sure thats possible as I have to remove the indicator from the previously selected item and put it on the new one.

Reply:
you'll need `System.Linq` for this one:
`ThumbContextMenu.Items.OfType<MenuItem>().ToList()`
store this somewhere and you have references to each menuitem. from there you cando `.ForEach(i => i.Icon = null)` and index to access a specific one for setting the icon

Reply:
I have changed it to 
```c#
if(!(item is MenuItem menuItem)) break;
menuItem.Icon = "";
if (menuItem.Header.ToString() == Type.ToString()) menuItem.Icon = this.Resources["SelectedIcon"];
```
which implements the other change you suggested, but still requires looping through the IEnumerable.

Reply:
if you reimplement the whole thing this way, the previous suggestion falls flat

Reply:
Ok, I've made `List<MenuItem> MenuItems` a global variable, and changed it to
```c#
MenuItems.ForEach(i => i.Icon = "");
MenuItems[(int)Type].Icon = this.Resources["SelectedIcon"];
```
Should be right

Reply:
lgtm, just remember to modify this if in the end we decide to put the delete button first in the context menu